metricflow/test/inference/context/test_snowflake.py

```diff
@@ -0,0 +1,122 @@
+    assert ctx_provider._column_type_from_show_columns_data_type("DATE") == SqlColumnType.DATETIME
+    assert ctx_provider._column_type_from_show_columns_data_type("TIMESTAMP_TZ") == SqlColumnType.DATETIME
+    assert ctx_provider._column_type_from_show_columns_data_type("TIMESTAMP_LTZ") == SqlColumnType.DATETIME
+    assert ctx_provider._column_type_from_show_columns_data_type("TIMESTAMP_NTZ") == SqlColumnType.DATETIME
+
+    # unknowns
+    assert ctx_provider._column_type_from_show_columns_data_type("BINARY") == SqlColumnType.UNKNOWN
+    assert ctx_provider._column_type_from_show_columns_data_type("TIME") == SqlColumnType.UNKNOWN
+
+
+def test_context_provider() -> None:  # noqa: D
+    # See for SHOW COLUMNS result dataframe spec:
+    # https://docs.snowflake.com/en/sql-reference/sql/show-columns.html
+    show_columns_result = pd.DataFrame(
+        {
+            "column_name": ["INTCOL", "STRCOL"],
+            "schema_name": ["SCHEMA", "SCHEMA"],
+            "table_name": ["TABLE", "TABLE"],
+            "database_name": ["DB", "DB"],
+            "data_type": [
+                json.dumps({"type": "FIXED", "nullable": False}),
+                json.dumps({"type": "TEXT", "nullable": True}),
+            ],
+        }
+    )
+
+    stats_result = pd.DataFrame(
+        {
+            "intcol_countdistinct": [10],
+            "intcol_min": [0],
+            "intcol_max": [10],
+            "intcol_countnull": [0],
+            "strcol_countdistinct": [40],
+            "strcol_min": ["aaaa"],
+            "strcol_max": ["zzzz"],
+            "strcol_countnull": [10],
+            "rowcount": [50],
+        }
+    )
+
+    client = MagicMock()
+    client.query = MagicMock()
+    client.query.side_effect = [show_columns_result, stats_result]
+
+    ctx_provider = SnowflakeInferenceContextProvider(
+        client=client,
+        tables=[SqlTable.from_string("db.schema.table")],
+        max_sample_size=50,
+    )
+
+    ctx = ctx_provider.get_context()
+
+    client.query.assert_has_calls(
+        [
+            call("SHOW COLUMNS IN TABLE db.schema.table"),
+            # make sure it produces correct SQL
+            call(
+                "SELECT "
+                "COUNT(DISTINCT intcol) AS intcol_countdistinct, "
+                "MIN(intcol) AS intcol_min, "
+                "MAX(intcol) AS intcol_max, "
+                # no need to query for count null if schema says it is not nullable
+                "0 AS intcol_countnull, "
+                "COUNT(DISTINCT strcol) AS strcol_countdistinct, "
+                "MIN(strcol) AS strcol_min, "
+                "MAX(strcol) AS strcol_max, "
+                "SUM(CASE WHEN strcol IS NULL THEN 1 ELSE 0 END) AS strcol_countnull, "
+                "COUNT(*) AS rowcount "
+                "FROM db.schema.table SAMPLE (50 ROWS)"
+            ),
+        ],
+        any_order=False,
+    )
```

Comment:
This is not a useful assertion. You're checking that a very specific call is being made, but this means we need to modify this whenever we change anything. Like if I decide to change "SUM(CASE...)" to "SUM(IF(....))" because that turns out to be faster or something, now I need to update this test. If I switch the formatting to use a jinja template or something with nicer line breaks, now I need to update this test.
```suggestion
```

Reply:
👍